src/types/assertionRef.ts

```diff
@@ -0,0 +1,59 @@
+import * as t from "io-ts";
+import { pipe } from "fp-ts/lib/function";
+import { JsonFromString } from "io-ts-types";
+
+import { LoginType } from "../utils/fastLogin";
+
+import { AssertionRef } from "../../generated/lollipop-api/AssertionRef";
+
+// LollipopData
+export type LollipopDataT = t.TypeOf<typeof LollipopData>;
+export const LollipopData = t.type({
+  assertionRef: AssertionRef,
+  loginType: LoginType,
+});
+
+// CompactLollipopData
+type CompactLollipopDataT = t.TypeOf<typeof CompactLollipopData>;
+const CompactLollipopData = t.type({
+  a: AssertionRef,
+  t: LoginType,
+});
+
+// LollipopDataFromCompact
+const LollipopDataFromCompact = new t.Type<
+  LollipopDataT,
+  CompactLollipopDataT,
+  CompactLollipopDataT
+>(
+  "LollipopDataFromCompact",
+  LollipopData.is,
+  (val, _context) =>
+    pipe(
+      val,
+      (data) => ({ assertionRef: data.a, loginType: data.t }),
+      LollipopData.decode
```

Comment:
Maybe is equivalent, but the error message could be more consistent if types composition will be needed.
```suggestion
      (data) => LollipopData.validate(data, _context)
```